src/bootstrap/builder.rs

```diff
@@ -590,6 +621,29 @@ impl<'a> Builder<'a> {
         StepDescription::run(v, self, paths);
     }
 
+    /// Print a command that will run the current step.
+    ///
+    /// This serves two purposes:
+    /// 1. Describe what step is currently being run.
+    /// 2. Describe how to run only this step in case it fails.
+    pub(crate) fn step_info(&self, step: &impl Step) {
+        if self.config.dry_run {
+            return;
+        }
+        print!(
+            "{} {} --stage {}",
+            // TODO: this is wrong, e.g. `check --stage 1` runs build commands first
```

Comment:
Not entirely sure how to fix this, but it seems critical to do so if we're to have some measure of success. I am hesitant to add additional annotations to the steps themselves, but perhaps that might be the way to go.